src/rhalphalib/sample.py

```diff
@@ -396,11 +410,13 @@ def getExpectation(self, nominal: bool = False, eval: bool = False):
                         raise NotImplementedError("per-bin effects for other nuisance parameter types")
                 else:
                     effect_down = self.getParamEffect(param, up=False)
-                    smoothStep = SmoothStep(param_scaled)
+                    smoothStep = SmoothStep(param) * param_scale
                     if param.combinePrior == "shape":
                         combined_effect = smoothStep * (1 + (effect_up - 1) * param_scaled) + (1 - smoothStep) * (1 - (effect_down - 1) * param_scaled)
                     elif param.combinePrior == "shapeN":
                         combined_effect = smoothStep * (effect_up**param_scaled) + (1 - smoothStep) / (effect_down**param_scaled)
+                    elif param.combinePrior == "shapeU":
+                        combined_effect = smoothStep * (effect_up**param_scaled) + (1 - smoothStep) / (effect_down**param_scaled)
```

Comment:
I suppose these three elifs could be merged into `elif param.combinePrior in ("shapeN", "shapeU", "lnN"):`